packages/proxy/test/unit/http/response-middleware.spec.ts

```diff
@@ -31,6 +31,36 @@ describe('http/response-middleware', function () {
     ])
   })
 
+  it('errors if this.next() is called more than once in the same middleware', function (done) {
+    const middleware = function () {
+      this.next()
+      this.next()
+    }
+
+    testMiddleware([middleware], {
+      onError (err) {
+        expect(err.message).to.equal('Error running proxy middleware: Cannot call this.next() more than once in the same middleware function. Doing so can cause unintended issues.')
+
+        done()
+      },
+    })
+  })
+
+  it('does not error if this.next() is called more than once in different middleware', function () {
+    const middleware1 = function () {
+      this.next()
+    }
+    const middleware2 = function () {
+      this.next()
+    }
+
+    return testMiddleware([middleware1, middleware2], {
+      onError () {
+        throw new Error('onError should not be called')
+      },
+    })
+  })
+
```

Comment:
Can we add some tests that exercise the `CopyCookiesFromIncomingRes` middleware to make sure the behavior we expect happens, like skipping if the experimental flag is off or setting the right header depending on the cookie or graceful failures?

Reply:
Added in [`c22269b` (#22320)](https://github.com/cypress-io/cypress/pull/22320/commits/c22269bb0d5bc43a1841b0a56de80b70e68f2550)